src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -210,11 +232,37 @@ void mf::WlSurface::frame(uint32_t callback)
 void mf::WlSurface::set_opaque_region(std::experimental::optional<wl_resource*> const& region)
 {
     (void)region;
+    // This isn't essential, but could enable optimizations
 }
 
 void mf::WlSurface::set_input_region(std::experimental::optional<wl_resource*> const& region)
 {
-    (void)region;
+    if (region)
+    {
+        // On Ubuntu 17.10 and later, the following two statements do the same thing.
+        // The former fails to compile on Ubuntu 16.04.
+
+        // pending.input_shape = WlRegion::from(region.value())->rectangle_vector();
```

Comment:
Hmm. I'm surprised this compiles. You're assigning a vector to an optional, optional vector and, by my reading, that's TWO implicit user-defined conversions (and one is the limit). I'd expect to need at least a couple of braces for deduced initialization.